sdk/src/env.rs

```diff
@@ -41,11 +41,21 @@ impl Env {
     // should be plumbed through this type with this type doing all RawVal
     // conversion.
 
-    pub fn put_contract_data<K: IntoTryFromVal, V: IntoTryFromVal>(&self, key: K, val: V) {
+    pub fn has_contract_data<K: IntoVal<Env, RawVal>>(&self, key: K) -> bool {
+        let rv = internal::Env::has_contract_data(self, key.into_val(self));
+        rv.try_into().unwrap()
+    }
+
+    pub fn get_contract_data<K: IntoVal<Env, RawVal>, V: IntoTryFromVal>(&self, key: K) -> V {
```

Comment:
I considered this when I originally made this change, but decided it wasn't a great idea. My perspective was that you should probably never be storing values that you can't convert in both directions, and doing so is almost surely a bug. This doesn't apply for keys, since you always provide them.